src/ccalltest.c

```diff
@@ -26,7 +27,7 @@ int __declspec(noinline)
 #else
 int __attribute__((noinline))
 #endif
-testUcharX(unsigned char x) {
+DLLEXPORT testUcharX(unsigned char x) {
```

Comment:
dll visibility matters for ccall - the default with mingw is only to export everything if there are no export attributes present. If you have a single dllexport that is explicitly specified, then only that function gets exported (unless you link with `-Wl,--export-all-symbols`). So it either needs to be nowhere or on everything you want to use.